voila/handler.py

```diff
@@ -118,12 +113,29 @@ async def get(self, path=None):
     def redirect_to_file(self, path):
         self.redirect(url_path_join(self.base_url, 'voila', 'files', path))
 
-    @tornado.gen.coroutine
-    def _jinja_kernel_start(self):
+    async def _jinja_kernel_start(self, nb):
         assert not self.kernel_started, "kernel was already started"
-        # See command above aboout not being able to start the kernel from a different thread
+        kernel_id = await self.kernel_manager.start_kernel(kernel_name=self.notebook.metadata.kernelspec.name, path=self.cwd)
```

Comment:
```suggestion
        kernel_id = await self.kernel_manager.start_kernel(kernel_name=nb.metadata.kernelspec.name, path=self.cwd)
```

Reply:
or insert `assert nb is self.notebook` to make this explicit.

Reply:
Actually `nb` is not needed in `_jinja_kernel_start`, I removed it.